query-node/substrate-query-framework/cli/src/generate/RelationshipGenerator.ts

```diff
@@ -0,0 +1,143 @@
+    field.relation = makeRelation('mtm', field.type, relatedField.name);
+    field.relation.joinTable = {
+      tableName: generateJoinTableName(currentObject.name, relatedObject.name),
+      joinColumn: generateJoinColumnName(currentObject.name),
+      inverseJoinColumn: generateJoinColumnName(relatedObject.name),
+    };
+    relatedField.relation = makeRelation('mtm', relatedField.type, field.name);
+
+    this.addRelatedImport(currentObject, relatedObject);
+    this.addToVisited(...[currentObject.name.concat(field.name), relatedObject.name.concat(relatedField.name)]);
+  }
+
+  addOne2Many(field: Field, relatedField: Field, currentObject: ObjectType, relatedObject: ObjectType): void {
+    field.relation = makeRelation('otm', field.type, relatedField.name);
+    relatedField.relation = makeRelation('mto', relatedField.type, field.name);
+
+    this.addRelatedImport(currentObject, relatedObject);
+    this.addToVisited(...[currentObject.name.concat(field.name), relatedObject.name.concat(relatedField.name)]);
+  }
+
+  addMany2One(field: Field, currentObject: ObjectType, relatedObject: ObjectType, relatedField: Field): void {
+    if (!relatedField.type) {
+      // Additinal field for field resolver
+      const fname = camelCase(currentObject.name).concat('s');
+      relatedField = new Field(fname, relatedObject.name, field.nullable, false, true);
+      relatedField.relation = makeRelation('otm', currentObject.name, field.name);
+      relatedObject.fields.push(relatedField);
+    } else {
+      relatedField.relation = makeRelation('otm', currentObject.name, field.name);
+    }
+
+    field.relation = makeRelation('mto', field.type, relatedField.name);
+
+    this.addRelatedImport(currentObject, relatedObject);
+    this.addToVisited(...[currentObject.name.concat(field.name), relatedObject.name.concat(relatedField.name)]);
+  }
+
+  addOne2One(field: Field, relatedField: Field, currentObject: ObjectType, relatedObject: ObjectType): void {
+    field.relation = makeRelation('oto', field.type, relatedField.name);
+    field.relation.joinColumn = true;
+    relatedField.relation = makeRelation('oto', relatedField.type, field.name);
+
+    this.addRelatedImport(currentObject, relatedObject);
+    this.addToVisited(...[currentObject.name.concat(field.name), relatedObject.name.concat(relatedField.name)]);
+  }
+
+  addToVisited(...args: string[]): void {
+    this._visited.push(...args);
+  }
+
+  isVisited(f: Field, o: ObjectType): boolean {
+    return this._visited.includes(o.name.concat(f.name));
+  }
+
+  addRelatedImport(o1: ObjectType, o2: ObjectType): void {
+    o1.relatedEntityImports.add(o2.name);
+    o2.relatedEntityImports.add(o1.name);
+  }
+
+  generate(): void {
+    this.model.types.forEach(currentObject => {
+      for (const field of currentObject.fields) {
+        if (this.isVisited(field, currentObject)) continue;
+
+        // ============= Case 1 =============
```

Comment:
Perhaps we can refactor Case1, Case2, Case3 into separate methods with self-describing names? Otherwise it's kinda hard to follow what these cases really differentiate.